### ISSUE TITLE
VS Code: JSON Preview shows form requesting API key and Submit does nothing.

### ISSUE THREAD

**Issue by robws** [Redocly/redocly-vs-code#66]
I pressed Preview while viewing a `swagger.json` file, and the preview screen showed "Want to see a context aware preview of your docs?" with a text box for API key. Adding my API key - even generating a new one - and pressing *Submit* had no effect. 

**What should have happened instead?**
Assuming pressing Submit would authenticate me with Redocly and show the OpenAPI Preview.

**Minimal reproducible OpenAPI snippet**
No snippet results in correct behavior; this does not appear to be a factor of the OpenAPI itself.

*If possible, include a short example of your OpenAPI definition that we can use to simulate the problem.*

**Screenshots**
<img width="668" alt="image" src="https://github.com/Redocly/redocly-vs-code/assets/29546533/6e317e4d-f9c8-4d28-b115-abac1b305ec8">

*If applicable, add screenshots to help explain your problem.*

**Additional context**

*Add any other information about the problem here, like: the extension version number; OpenAPI version you're using in your definition; your operating system.*
Latest extension, Mac, VS Code latest version.

**Comment by tatomyr** [comment]
That's strange. Could you check if logging via Redocly CLI helps here? For that, please run `npx @redocly/cli login`, then close and reopen your editor. Then try the 'Preview' feature again.

**Comment by robws** [comment]
Where do I run that from, and will it leave anything in my system I don't expect? When I first ran it it seemed to put something in NPM.

It says "Need to install the following packages:
  @redocly/cli@1.12.0"
  
  What does that do?

**Comment by robws** [comment]
I see this:
<img width="975" alt="image" src="https://github.com/Redocly/redocly-vs-code/assets/29546533/e8af9644-0dd8-4886-aaf1-17acf1d6d816">

Which is bogus, because you can't literally get your API key from that address (you would have had to have retrieved it when you created it)

And pasting it doesn't result in anything other than "logging in..." then a return to the prompt.

**Comment by robws** [comment]
Restarting VS Code has the same behavior as described.

**Comment by tatomyr** [comment]
I see. It looks like you're using an expired or revoked key. The message should look like the following:

<img width="575" alt="Screenshot 2024-05-01 at 08 20 18" src="https://github.com/Redocly/redocly-vs-code/assets/11423792/0a289f8f-2a57-4f76-a16e-fce0d3829e33">


Please follow the link and generate a new key. 

This can also explain why you cannot log in directly with the extension. This is a bug, indeed. There should be a warning in case of a wrong key.


> It says "Need to install the following packages:
> @redocly/cli@1.12.0"
>
> What does that do?

It installs [Redocly CLI](https://www.npmjs.com/package/@redocly/cli) (which the extension is driven by) in your local NPM cache.

**Comment by robws** [comment]
Thank you! I'll make another key and give it a shot. I was confused why the preview window didn't seem to do anything when I pressed submit. I was watching and developer tools to see if there were any network traffic, and didn't see any JS errors. But I guess there wouldn't be if you coded it to trap errors! 

I wasn't sure about whether or not I pasted the right thing into the CLI, because of course it doesn't show you the text of what you paste and sometimes you're not sure if you put other characters into that prompt at all.

**Comment by robws** [comment]
It is indeed our VPN, Cloudflare Warp Zero-Trust, as the authorization works fine when I turn it off, as does the extension to preview.

I didn't realize this at first, as the preview was failing, and possibly has multiple reasons for failure. I'm assuming when it can't reach the redocly servers, it perpetually shows the "Log in" form screen.

**Comment by tatomyr** [comment]
> I'm assuming when it can't reach the redocly servers, it perpetually shows the "Log in" form screen.

Yes. It relies on authenticating through a Redocly endpoint, so the VPN have to be configured appropriately for the extension to be able to verify the token.

**Comment by robws** [comment]
Once the token is verified, does any more communication happen? 

I suspect the company would probably want to know if API specs must be sent to our stored at redocly, or if the preview tool can be used otherwise.

**Comment by tatomyr** [comment]
The extension tries to verify the token each time it gets loaded. So (more or less) each time you reopen the editor it will send a request to the Redocly API.

**Comment by romanpryshliak** [comment]
Currently, the extension doesn't accept the personal API key for me. However, I managed to get it working with an organization API key from Redocly Workflows: ![Image](https://github.com/user-attachments/assets/42bf8033-678f-4da5-9a19-e087a26542fb)

**Comment by leodolbir** [comment]
I observed the problem described above and unable to use the preview. I was however able to login through the CLI, so I'm confident the VPN has nothing to do with it. Also looking through developer tools there are input fields and the button but no form to submit or a javascript handler
![Image](https://github.com/user-attachments/assets/53a9efb1-7e82-4bda-87ff-ea94614c752e)

**Comment by jmehnle** [comment]
I, too, am suffering from this problem: I paste my personal API key into the "API key" input field in the preview tab and click "Submit", yet absolutely nothing happens. I can click the button many times, to no avail. I have the `redocly` CLI installed via `npm`, and I can run it on the shell.

**Comment by tatomyr** [comment]
No longer relevant since the login flow has been changed in [v0.6.0](https://open-vsx.org/extension/Redocly/openapi-vs-code/changes#0.6.0).
